Source/Charts/Charts/ChartViewBase.swift

```diff
@@ -39,43 +39,51 @@ open class ChartViewBase: NSUIView, ChartDataProvider, AnimatorDelegate
 {
     // MARK: - Properties
     
-    /// - returns: The object representing all x-labels, this method can be used to
-    /// acquire the XAxis object and modify it (e.g. change the position of the
-    /// labels)
-    @objc open var xAxis: XAxis
-    {
-        return _xAxis
-    }
-    
     /// The default IValueFormatter that has been determined by the chart considering the provided minimum and maximum values.
     internal var _defaultValueFormatter: ValueFormatter? = DefaultValueFormatter(decimals: 0)
-
-    /// object that holds all data that was originally set for the chart, before it was modified or any filtering algorithms had been applied
-    internal var _data: ChartData?
     
+    /// The data for the chart
+    @objc open var data: ChartData? {
+        didSet {
```

Comment:
does `didSet` have any advantage over the old setter? Or just `didSet` describe the job more properly?

Reply:
I don't understand. The old way had two variables, one stored and one computed. This just has the one stored with a property observer to do the calculations.

Reply:
I mean setter vs. didSet.